packages/remix-react/components.tsx

```diff
@@ -1049,21 +1048,28 @@ export function useFormAction(
   let location = useLocation();
   let { search, hash } = resolvedPath;
   let isIndexRoute = id.endsWith("/index");
+  let params = new URLSearchParams(search);
 
   if (action == null) {
     search = location.search;
     hash = location.hash;
+    params = new URLSearchParams(search);
 
     // When grabbing search params from the URL, remove the automatically
     // inserted ?index param so we match the useResolvedPath search behavior
     // which would not include ?index
     if (isIndexRoute) {
-      let params = new URLSearchParams(search);
       params.delete("index");
-      search = params.toString() ? `?${params.toString()}` : "";
     }
   }
 
+  // Emulate additional methods when JavaScript is missing/disabled/pending
+  method = method.toLowerCase() as FormMethod;
+  if (["put", "patch", "delete"].includes(method)) {
+    params.set("_method", method);
+  }
+
+  search = params.toString() ? `?${params.toString()}` : "";
```

Comment:
Is this also affecting the fetcher.Form component? AFAIK the fetcher.Form works without JS too since it renders a normal form tag

Reply:
Yes, `fetcher.Form` and `Form` are backed by the same `FormImpl`+`useFormAction`